### ISSUE TITLE
Extend to become a Ni-Fi library

### ISSUE THREAD

**Issue by Thesola10** [Thesola10/PictoChat#4]
Okay, this is _very far-fetched_ but since I suspect the PictoChat procotol to depend on Ni-Fi sessions (like DS Multiplayer games), this library could become a starting point to reverse-engineering all of the DS's Ni-Fi protocols (imagine, homebrew Download Play games from PC) if it becomes developed enough.

**Comment by Thesola10** [comment]
Here's an idea: a PC game which uses Nintendo DS as controllers, via Download Play and DS Multiplayer. That would be so cool!

**Comment by Krutonium** [comment]
Sorry to punch a hole in one of your hopes here

Without something similar to the Wii's Truncha Bug, there is no way to send an unsigned (by Nintendo) binary to the NDS via DSDLP. At least on the stock Firmware. You can remove that check using CFW, but that's incredibly risky and involves shorting part of your DS's board to enable RW on the firmware chip.

That being said, for Ni-Fi, I can see that working.

**Comment by Thesola10** [comment]
> there is no way to send an unsigned binary to the NDS via DSDLP

Yeah, figures. I remember having read something like that before.

~~But couldn't [HaxxStation](https://github.com/Gericom/dspatch) help?~~

**Comment by Krutonium** [comment]
As far as I know, that's still serving a ROM signed by Nintendo
